cmd/contour/serve.go

```diff
@@ -316,13 +316,29 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 	// step 11. register leadership election.
 	eventHandler.IsLeader = setupLeadershipElection(&g, log, ctx, clients, eventHandler.UpdateNow)
 
+	updateChannel := make(chan k8s.StatusUpdate)
+	sh := k8s.StatusUpdateHandler{
+		Log:           log.WithField("context", "StatusUpdateWriter"),
+		Clients:       clients,
+		UpdateChannel: updateChannel,
+		LeaderElected: eventHandler.IsLeader,
+		Converter:     converter,
+	}
+
+	suw := k8s.StatusUpdateWriter{
+		UpdateChannel: updateChannel,
+	}
```

Comment:
We can remove the need to manage a channel in the caller by adding a helper to the handler:
```Go
func (*k8sStatusUpdateHandler) Writer() k8s.StatusUpdater
```

The update handler can create the channel implicitly and hand one end to the writer, but the caller doesn't need to know there are channels involved and you don't need to expose StatusUpdateWriter as a type.